File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/BooleanFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive;

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;
import io.github.oleksivio.telegram.bot.api.model.annotation.BooleanState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface BooleanFilter {

    AnnotationState status() default AnnotationState.ON;

    BooleanState state() default BooleanState.NON_NULL;

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/BooleanFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState
import io.github.oleksivio.telegram.bot.api.model.annotation.BooleanState

/**
 * [BooleanFilter] annotation used for filtering incoming
 * [Boolean] value.
 *
 * @param state [BooleanState] is enum filtering condition
 * it could be TRUE, FALSE and NON_NULL
 */
@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class BooleanFilter(
        val status: AnnotationState = AnnotationState.ON,
        val state: BooleanState = BooleanState.NON_NULL
)
